mmengine/infer/infer.py

```diff
@@ -0,0 +1,491 @@
+        which will be used in __call__ like this:
+
+        .. code-block:: python
+
+            def __call__(self, inputs, batch_size=1, **kwargs):
+                chunked_data = self.preprocess(inputs, batch_size, **kwargs)
+                for batch in chunked_data:
+                    preds = self.forward(batch, **kwargs)
+
+        Args:
+            inputs (InputsType): Inputs given by user.
+            batch_size (int): batch size. Defaults to 1.
+
+        Yields:
+            Any: Data processed by the ``pipeline`` and ``collate_fn``.
+        """
+        if isinstance(inputs, str):
+            backend = get_file_backend(inputs)
+            # Backends like HttpsBackend do not implement `isdir`, so only
+            # those backends that implement `isdir` could accept the inputs
+            # as a directory
+            if hasattr(backend, 'isdir') and isdir(inputs):
+                filename_list = list_dir_or_file(inputs, list_dir=False)
+                inputs = [
+                    join_path(inputs, filename) for filename in filename_list
+                ]
+            else:
+                inputs = [inputs]
+
+        if not isinstance(inputs, list):
+            inputs = [inputs]
+
+        chunked_data = self._get_chunk_data(
+            map(self.pipeline, inputs), batch_size)
+        yield from map(self.collate_fn, chunked_data)
+
+    @torch.no_grad()
+    def forward(self, inputs: Union[dict, tuple], **kwargs) -> Any:
+        """Feed the inputs to the model."""
+        return self.model.test_step(inputs)
+
+    @abstractmethod
+    def visualize(self,
+                  inputs: list,
+                  preds: Any,
+                  show: bool = False,
+                  **kwargs) -> List[np.ndarray]:
+        """Visualize predictions.
+
+        Customize your visualization by overriding this method. visualize
+        should return a visualization result, which could be np.ndarray or any
+        other objects
+
+        Args:
+            inputs (list): Inputs preprocessed by :meth:`preprocess`.
+            preds (Any): Predictions of the model.
+            show (bool): Whether to display the image in a popup window.
+                Defaults to False.
+
+        Returns:
+            List[np.ndarray]: Visualization results.
+        """
+
+    @abstractmethod
+    def postprocess(
+        self,
+        preds: Any,
+        visualization: List[np.ndarray],
+        return_datasample=False,
+        **kwargs,
+    ) -> dict:
+        """Postprocess predictions.
+
+        Customize your postprocess by overriding this method. postprocess
+        should return a dict with visualization results and inference results.
+
+        Args:
+            preds (List[Dict]): Predictions of the model.
+            visualization (np.ndarray): Visualized predictions.
+            return_datasample (bool): Whether to return results as datasamples.
```

Comment:
This will change the type of `predictions`?

Reply:
Yes, the prediction could be transferred from DataSample to dict